org-roam.el

```diff
@@ -135,7 +135,24 @@ space-delimited strings.
   :group 'org-roam)
 
 (defcustom org-roam-tag-sources '(prop)
-  "Sources to obtain tags from."
+  "Sources to obtain tags from.
+
+It should be a list of symbols representing any of the following
+extraction methods:
+
+1. 'prop: This extracts tags from the =#+ROAM_TAGS=
+   property.  Tags are space delimited, and can be multi-word
+   using double quotes.
+
+2. 'all-directories: All sub-directories relative to
+   =org-roam-directory= are extracted as tags.  That is, if a file
+   is located at relative path =foo/bar/file.org=, the file will
+   have tags =foo= and =bar=.
+
+3. 'last-directory: Extracts the last directory relative to
+   =org-roam-directory= as the tag.  That is, if a file is located
+   at relative path =foo/bar/file.org=, the file will have tag
+   =bar=."
```

Comment:
I think it would be a good idea to add info on how users can add their own sources in this documentation.

e.g. You can add your own source `'foo` by defining the function `org-roam--extract-tags-foo`

Reply:
#blamejethro 😄
I'm off to bed now, I'll check this in the morning.